18F/identity-idp
# LG-9440 Add headers & data on verify_info w/ tests

## 🎫 Ticket
[LG-9440](https://cm-jira.usa.gov/browse/LG-9440) Apply all DAV functional and design changes to the controller-based version of the verify info page

## 🛠 Summary of changes
All items in the list were added to `/verify_info` view when `in_person_verify_info_controller_enabled = true` AND `in_person_capture_secondary_id_enabled = true`

1. State-issued ID header
2. Current residential address header
3. Social Security Number header
4. Added Issuing state below Date of birth (state_id_jurisdiction)
5. Added a space after ID number
6. Added Address line 1 (identity_doc_address1)
7. Added Address line 2 (identity_doc_address2)
8. Added City (identity_doc_city)
9. Added State (identity_doc_address_state)
10. Added Zip Code (identity_doc_zipcode)

## 📜 Testing Plan
* In all cases, confirm **visual** and **functional behavior** observed on `/verify` has been applied to `/verify_info`.
* When I compared the two I looked at headers, data, redirects (Cancel, Update x3, Continue), Checked/Unchecked boxes, responsiveness, and flows. I created a [document](https://docs.google.com/document/d/17uDrwBdsPtT57u2EvY_SVv5slZE8CGoe7amJ6QhVMdA/edit#) to compare /verify and /verify_info in many aspects to record my observations. 
* The Summary of changes listed above are the differences I observed. 
*  **When you are testing please confirm that all visual and behavioral changes have been made and that I did not miss any.**

1. Set `in_person_verify_info_controller_enabled` and `in_person_capture_secondary_id_enabled` in config/application.yml to the test case you are testing below.
2. Create an account
3. Go to `/verify`
4. Upload state-issued ID (front + back) to fail ID verification 
5. Go to in person proofing, fill out steps to get to Verify your information
6. Open another tab
7. One tab should use url `localhost:3000/verify/in_person/verify` and the second tab should use url `localhost:3000/verify/in_person/verify_info`
8.  Observe visual and behavior

- [ ] Test Case 1: in_person_verify_info_controller_enabled = true AND in_person_capture_secondary_id_enabled: true
* Confirm you are taken to '/verify_info' in this flow
*  ⭐  ⭐  `/verify_info` should be updated to look and behave like `/verify`  👈  **Task of this issue** ⭐  ⭐ 

- [ ] Test Case 2: in_person_verify_info_controller_enabled = true AND in_person_capture_secondary_id_enabled: false
* This is all existing behavior. This check is just to confirm existing behavior is not effected
* Confirm you are taken to '/verify_info' in this flow and you do not get a 404
* Confirm state-ID info is not displayed on `/verify_info' (double address info)

- [ ] Test Case 3: in_person_verify_info_controller_enabled = false AND in_person_capture_secondary_id_enabled: true
* This is all existing behavior. This check is just to confirm existing behavior is not effected
* Confirm you are taken to `/verify` in this flow
* Confirm state-ID info is displayed on `/verify` (double address info)
* Confirm you get an error if you try to go to `/verify_info`

- [ ] Test Case 4: in_person_verify_info_controller_enabled = false AND in_person_capture_secondary_id_enabled: false
* This is all existing behavior. This check is just to confirm existing behavior is not effected
* Confirm you are taken to `/verify` in this flow
* Confirm state-ID info is not displayed on `/verify_info' (double address info)
* Confirm you get an error if you try to go to `/verify_info`

## 👀 Screenshots

<details>
<summary>Test Case 1: in_person_verify_info_controller_enabled = true AND in_person_capture_secondary_id_enabled: true</summary>

**before**
<img width="1402" alt="case 1 before" src="https://github.com/18F/identity-idp/assets/125507397/aa54d403-d5a7-45c9-b0ee-52a381e2c218">

**after**
<img width="1451" alt="test case 1 after" src="https://user-images.githubusercontent.com/125507397/236939261-f7b08c4f-52ac-4bf3-b597-739f00fb07eb.png">
</details>

<details>
<summary>Test Case 2: in_person_verify_info_controller_enabled = true AND in_person_capture_secondary_id_enabled: false</summary>

**before**
<img width="1295" alt="case 2 before" src="https://github.com/18F/identity-idp/assets/125507397/fb09baa2-1649-49c1-aa5b-68478c1e9d52">

**after**
<img width="1265" alt="test case 2 after" src="https://user-images.githubusercontent.com/125507397/236940724-f49af1e8-fb06-4536-b9e4-0aee10e4c479.png">
</details>

<details>
<summary>Test Case 3: in_person_verify_info_controller_enabled = false AND in_person_capture_secondary_id_enabled: true</summary>

**before**
<img width="1279" alt="case 3 before" src="https://github.com/18F/identity-idp/assets/125507397/39d953e3-124a-4322-b8e9-85ff5f811d41">

**after**
<img width="1305" alt="test case 3 after" src="https://user-images.githubusercontent.com/125507397/236941923-486b0456-8ab9-43ed-a6eb-05b72ef27e3c.png">
</details>

<details>
<summary>Test Case 4: in_person_verify_info_controller_enabled = false AND in_person_capture_secondary_id_enabled: false</summary>

**before**
<img width="1232" alt="case 4 before" src="https://github.com/18F/identity-idp/assets/125507397/30a4734a-b9c3-414c-b040-96844c039005">

**after**
<img width="1300" alt="case 4 after" src="https://user-images.githubusercontent.com/125507397/236944327-98469c3f-7593-4d85-a539-c945eea4a09f.png">
</details>
